### PR TITLE
Add pull_request_review_thread Event Support

### PR DESCRIPTION
**Pull request by tylerbwong**
GitHub added a new `pull_request_review_thread` webhook event that gets sent every time a comment is resolved or unresolved.

### CODE REVIEW

File: octohook/events.py [hunk 1]
```diff
@@ -552,6 +552,22 @@ def __init__(self, payload: dict):
         self.changes = _optional(payload, "changes", RawDict)
 
 
+class PullRequestReviewThreadEvent(BaseWebhookEvent):
+    """
+    https://developer.github.com/v3/activity/events/types/#pullrequestreviewthreadevent
```
Comment (tylerbwong) [inline octohook/events.py:557]: There are no docs up yet for this event, but https://github.com/google/go-github/issues/2324#issuecomment-1080659310 states that GitHub has an internal ticket to update the docs.

**Comment by tylerbwong** [comment]
@doodla I had to remove `reactions` from the `Comment` payload since tests were failing due to "+1"/"-1" not matching up to the property names I used "plus_one"/"minus_one".

```json
"reactions": {
    "url": "https://api.github.com/repos/doodla/octohook-playground/pulls/comments/839046410/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
}
```

File: octohook/events.py [hunk 2]
```diff
@@ -41,21 +41,22 @@
...
     Enterprise,
+    Thread,
 )
 
 
 class BaseWebhookEvent:
     payload: dict
     action: Optional[str] = None
-    sender: User
+    sender: Optional[User]
```
Comment (tylerbwong) [inline octohook/events.py:51]: This was not present in the "unresolved" payload.

File: octohook/models.py [hunk 3]
```diff
@@ -253,6 +253,10 @@ class Repository(BaseGithubModel):
     public: Optional[bool]
     master_branch: Optional[str]
     permissions: Optional[Permissions]
+    allow_forking: Optional[bool]
+    is_template: Optional[bool]
+    topics: List[str]
```
Comment (doodla) [inline octohook/models.py:258]: This is optional in the `__init__` block.

**Comment by doodla** [comment]
> @doodla I had to remove `reactions` from the `Comment` payload since tests were failing due to "+1"/"-1" not matching up to the property names I used "plus_one"/"minus_one". If we want to add reactions to the model we might have to make the tests less strict.
> 
> ```json
> "reactions": {
>     "url": "https://api.github.com/repos/doodla/octohook-playground/pulls/comments/839046410/reactions",
>     "total_count": 0,
>     "+1": 0,
>     "-1": 0,
>     "laugh": 0,
>     "hooray": 0,
>     "confused": 0,
>     "heart": 0,
>     "rocket": 0,
>     "eyes": 0
> }
> ```

Ah that's interesting that github sends numbers in the payload. Can we add it as a `RawDict` instead?

File: octohook/models.py (hunk 3 above)
Comment (tylerbwong) [reply octohook/models.py:258]: I have it default to an empty list, but I changed it to just `payload.get("topics").

**Comment by tylerbwong** [comment]
> > @doodla I had to remove `reactions` from the `Comment` payload since tests were failing due to "+1"/"-1" not matching up to the property names I used "plus_one"/"minus_one". If we want to add reactions to the model we might have to make the tests less strict.
> > ```json
> > "reactions": {
> >     "url": "https://api.github.com/repos/doodla/octohook-playground/pulls/comments/839046410/reactions",
> >     "total_count": 0,
> >     "+1": 0,
> >     "-1": 0,
> >     "laugh": 0,
> >     "hooray": 0,
> >     "confused": 0,
> >     "heart": 0,
> >     "rocket": 0,
> >     "eyes": 0
> > }
> > ```
> 
> Ah that's interesting that github sends numbers in the payload. Can we add it as a `RawDict` instead?

Ah didn't realize we could do that. Just added :+1:

**Review by doodla — APPROVED** [review]

**Pull request merged**